zztboard.js

```diff
@@ -147,20 +135,34 @@ ZZTBoard.prototype.moveActor = function(actorIndex, x, y)
    var srcTile = this.get(actorData.x, actorData.y);
    var dstTile = this.get(x, y);
 
+   // set the underTile to where the actor was
    this.set(actorData.x, actorData.y, actorData.underTile);
+
+   // set the new location for actor
    this.set(x, y, srcTile);
 
    actorData.x = x;
    actorData.y = y;
+
+   // Only apply this to tiles that can exist under the actor
+   // - e.g., passages
+   // - not torches; those are picked up
+   if (dstTile?.properties?.allowUndertile)
+     actorData.underTile = dstTile
+   else
+     actorData.underTile = _ZZTBoard_BoardEmpty
```

Comment:
Only assign the dst tile to the player's underTile if the tile is assignable as such.  The player's underTile must be cleared out after otherwise the player will leave a trail of objects as it moves.